runtime/appruntime/metrics/datadog/datadog.go

```diff
@@ -0,0 +1,188 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/rs/zerolog"
+
+	"encore.dev/appruntime/config"
+	"encore.dev/appruntime/metadata"
+	"encore.dev/appruntime/metrics/system"
+	"encore.dev/metrics"
+)
+
+func New(svcs []string, cfg *config.DatadogProvider, meta *metadata.ContainerMetadata, rootLogger zerolog.Logger) *Exporter {
+	// Precompute container metadata labels.
+	return &Exporter{
+		svcs: svcs,
+		cfg:  cfg,
+		containerMetadataLabels: []string{
+			"service_id:" + meta.ServiceID,
+			"revision_id:" + meta.RevisionID,
+			"instance_id:" + meta.InstanceID,
+		},
+		rootLogger: rootLogger,
+	}
+}
+
+type Exporter struct {
+	svcs                    []string
+	cfg                     *config.DatadogProvider
+	containerMetadataLabels []string
+	rootLogger              zerolog.Logger
+}
+
+func (x *Exporter) Shutdown(_ context.Context) {}
+
+func (x *Exporter) Export(ctx context.Context, collected []metrics.CollectedMetric) error {
+	now := time.Now()
+	data := x.getMetricData(now, collected)
+	data = append(data, x.getSysMetrics(now)...)
+	body := datadogV2.MetricPayload{Series: data}
+
+	ctx = x.newContext(ctx)
+	configuration := datadog.NewConfiguration()
+	apiClient := datadog.NewAPIClient(configuration)
+	api := datadogV2.NewMetricsApi(apiClient)
```

Comment:
Do we need to instantiate the client at export time? Intuitively I would expect that to happen during Exporter initialization.

Reply:
We don't, I'll change that 👍